pvac_seq/binding_filter.py

```diff
@@ -0,0 +1,189 @@
+import argparse
+import sys
+import re
+import os
+import csv
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('input', type=argparse.FileType('r'),
+                        help="Input list of variants")
+    parser.add_argument('fof', type=argparse.FileType('r'),
+                        help="FOF containing list of parsed epitope files " +
+                        "for different allele-length combinations (same sample)")
+    parser.add_argument('output', type=argparse.FileType('w'),
+                        help="Output .xls file containing list of filtered " +
+                        "epitopes based on binding affinity for each " +
+                        "allele-length combination per gene")
+    parser.add_argument('-c', '--fold-change', type=int,
+                        help="Minimum fold change between mutant binding " +
+                        "score and wild-type score. The default is 0, which " +
+                        "filters no results, but 1 is often a sensible " +
+                        "default (requiring that binding is better to the MT " +
+                        "than WT)",
+                        default=0,
+                        dest="minimum_fold_change")
+    parser.add_argument('-b', '--binding-threshold', type=int,
+                        help="Report only epitopes where the mutant allele " +
+                        "has ic50 binding scores below this value; default 500",
+                        default=500,
+                        dest="binding_threshold")
+
+    args = parser.parse_args()
+
+    #precompile regex patterns used later
+    chromosome_name = re.compile(r'^chromosome_name')
+    netmhc_subber = re.compile(r"_netmhc")
+
+    #open the variants file and parse into variants dictionary
+    reader = csv.reader(args.input, delimiter='\t')
+    variants = {}
+
+    for data in reader:
+        if chromosome_name.match(data[0]):
+            variants['header'] = '\t'.join(data)
+            continue
+
+        #1	chromosome_name
+        #2	start
+        #3	stop
+        #4	reference
+        #5	variant
+        #6	gene_name
+        #7	transcript_name
+        #8	amino_acid_change
+        #9	ensembl_gene_id
+        #10	wildtype_amino_acid_sequence
+
+        gene = data[5]
+        amino_acid_change = data[7]
+        variants[ gene + '\t' + amino_acid_change] = '\t'.join(data)
+
+    #dump header data to output file
+    args.input.close()
+    if 'header' not in variants:
+        # usage("Header not defined in variant input file")
+        parser.print_help()
+        print("\nError: Header not defined in variant input file")
+        exit(1)
+    args.output.write('\t'.join([
```

Comment:
We might also want to use the csv.writer here. One thing to note when using csv.writer is that I had to overwrite the `lineterminator` because otherwise it's using a slightly different newline character which causes the file diff to fail.